app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt

```diff
@@ -569,6 +572,214 @@ class PatientRepositoryAndroidTest {
+    assertThat(savedPatient.searchableName).isEqualTo("NewName")
+    assertThat(savedPatient.gender).isEqualTo(Gender.TRANSGENDER)
+  }
+
+  @Test
+  fun when_phone_number_is_updated_it_should_be_saved() {
+    val addressToSave = testData.patientAddress()
+
+    val originalSavedPatient = testData.patient(
+        syncStatus = SyncStatus.DONE,
+        addressUuid = addressToSave.uuid
+    )
+
+    val patientProfile = PatientProfile(
+        patient = originalSavedPatient,
+        address = addressToSave,
+        phoneNumbers = listOf(
+            testData.patientPhoneNumber(
+                patientUuid = originalSavedPatient.uuid,
+                number = "111111111",
+                phoneType = PatientPhoneNumberType.LANDLINE,
+                createdAt = Instant.now(clock),
+                updatedAt = Instant.now(clock)
+            ),
+            testData.patientPhoneNumber(
+                patientUuid = originalSavedPatient.uuid,
+                number = "2222222222",
+                phoneType = PatientPhoneNumberType.MOBILE,
+                createdAt = Instant.now(clock),
+                updatedAt = Instant.now(clock)
+            )
+        )
+    )
+
+    patientRepository.save(listOf(patientProfile))
+        .blockingAwait()
+
+    val updatedAfter = Duration.ofDays(1L)
+    (clock as TestClock).advanceBy(updatedAfter)
+
+    val phoneNumberToUpdate = patientProfile.phoneNumbers[1].copy(number = "12345678", phoneType = PatientPhoneNumberType.LANDLINE)
+
+    patientRepository.updatePhoneNumberForPatient(originalSavedPatient.uuid, phoneNumberToUpdate).blockingAwait()
+
+    val phoneNumbersSaved = database.phoneNumberDao().phoneNumber(originalSavedPatient.uuid).firstOrError().blockingGet()
+
+    val phoneNumber = phoneNumbersSaved.find { it.uuid == phoneNumberToUpdate.uuid }
+
+    assertThat(phoneNumber).isNotNull()
+    assertThat(phoneNumber!!.number).isEqualTo("12345678")
+    assertThat(phoneNumber.phoneType).isEqualTo(PatientPhoneNumberType.LANDLINE)
+    assertThat(phoneNumber.updatedAt).isEqualTo(patientProfile.phoneNumbers[1].updatedAt.plus(updatedAfter))
+    assertThat(phoneNumber.updatedAt).isNotEqualTo(phoneNumber.createdAt)
+  }
+
+  @Test
+  fun phone_number_should_be_saved_properly() {
+    val addressToSave = testData.patientAddress()
+
+    val originalSavedPatient = testData.patient(
+        syncStatus = SyncStatus.DONE,
+        addressUuid = addressToSave.uuid
+    )
+
+    val patientProfile = PatientProfile(
+        patient = originalSavedPatient,
+        address = addressToSave,
+        phoneNumbers = listOf(
+            testData.patientPhoneNumber(
+                patientUuid = originalSavedPatient.uuid,
+                number = "111111111",
+                phoneType = PatientPhoneNumberType.LANDLINE,
+                createdAt = Instant.now(clock),
+                updatedAt = Instant.now(clock)
+            )
+        )
+    )
```

Comment:
This is an opinion, but I prefer the dangling brackets since it tells me where the scope ends. Otherwise, I have to open it an IDE and place a cursor there. But I'll change this for now since this is what we are using everywhere.